src/meta/src/hummock/manager/mod.rs

```diff
@@ -224,13 +224,39 @@ impl HummockManager {
         // Skip this check in e2e compaction test, which needs to start a secondary cluster with
         // same bucket
         if !deterministic_mode {
-            write_exclusive_cluster_id(
+            let use_new_object_prefix_strategy = write_exclusive_cluster_id(
                 state_store_dir,
                 env.cluster_id().clone(),
                 object_store.clone(),
             )
             .await?;
-
+            if use_new_object_prefix_strategy {
+                match env.system_params_manager_impl_ref() {
+                    SystemParamsManagerImpl::Kv(mgr) => {
+                        mgr.set_param("use_new_object_prefix_strategy", Some("true".to_owned()))
+                            .await
+                            .unwrap();
+                    }
+                    SystemParamsManagerImpl::Sql(mgr) => {
+                        mgr.set_param("use_new_object_prefix_strategy", Some("true".to_owned()))
+                            .await
+                            .unwrap();
+                    }
+                };
+            } else {
+                match env.system_params_manager_impl_ref() {
+                    SystemParamsManagerImpl::Kv(mgr) => {
+                        mgr.set_param("use_new_object_prefix_strategy", Some("false".to_owned()))
+                            .await
+                            .unwrap();
+                    }
+                    SystemParamsManagerImpl::Sql(mgr) => {
+                        mgr.set_param("use_new_object_prefix_strategy", Some("false".to_owned()))
+                            .await
+                            .unwrap();
+                    }
+                };
+            }
```

Comment:
I suggest we move the logic to [`MetaSrvEnv::new`](https://github.com/risingwavelabs/risingwave/blob/380912047757c25628e82c69e47a7fe8ef7a4c6d/src/meta/src/manager/env.rs#L361) instead here becuase:
- `cluster_first_launch` can be obtained directly there without interacting with object store and modifying `write_exclusive_cluster_id`
- We can avoid making `use_new_object_prefix_strategy` mutable and instead pass the init value for `use_new_object_prefix_strategy` to init system params manager.

Reply:
> cluster_first_launch can be obtained directly there

For `KV` meta store, it is, but for `SQL` meta store, can we know whether it is the first launch here?

Reply:
cc @yezizp2012  Can we also judge cluster_first_launch for SQL backend, just like KV backend? 
Otherwise we need to modify this system variable later, but "_Making the use_new_object_prefix_strategy mutable will be dangerous because user can techinically change it via ALTER in SQL and the after a restart, the whole cluster will crash._"

Reply:
After some offline discussion, for SQL backend I use `MigratorTrait::get_applied_migrations()` to check whether it's a new cluster:
- if `m20230908_072257_init ` is applied, it's an old cluster
- if `m20230908_072257_init ` is not applied, it's a new cluster.

Then we can make `use_new_object_prefix_strategy` immutable.